### PR TITLE
limit account nonce

### LINKED ISSUE #3593: limit account nonce to 2^64-1
### What would you like to be added
limit account nonce to 2^64-1

### Why is this needed
see https://github.com/ethereum/go-ethereum/commit/f32feeb260054d7c90c1efd57dfec6b717717e18

### How important you think this is for IoTeX
- [] must have
- [x] should have
- [] nice to have

### Additional information
this is EIP-2681 requirements, at this point no account/address has nonce value even close to this upper limit. So this is not urgent, can be done at any time

### PR DESCRIPTION
**Pull request by huangzhiran**
# Description
limit account nonce to 2^64-1 (my implement limit to 2^64-2)

Fixes #3593

## Type of change
- [x] Code refactor or improvement

# How Has This Been Tested?
- [x] make test

**Test Configuration**:
- Firmware version: macOS Monterey arm64
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: state/account.go [hunk 1]
```diff
@@ -139,8 +141,16 @@ func (st *Account) AccountType() int32 {
 	return st.accountType
 }
 
+func (st *Account) isNonceOverflow() bool {
+	// if st.accountType is 0, nonce will add 2. so use +2 as overflow flag
+	return st.nonce+2 < st.nonce
```
Comment (huof6829) [inline state/account.go:146]: Limit account nonce to be between 0 and 2^64-1

Comment (huangzhiran) [reply state/account.go:146]: for legacy account, nonce will finally +2 ( when SetPendingNonce() and PendingNonce() ), so maybe nonce in [0, 2^64-2] will be better

File: state/account.go [hunk 2]
```diff
@@ -139,8 +141,16 @@ func (st *Account) AccountType() int32 {
...
 }
 
+func (st *Account) isMaxNonce() bool {
+	// for legacy account, nonce will finally +2. so use +2 as overflow flag
+	return st.nonce+2 < st.nonce
+}
+
 // SetPendingNonce sets the pending nonce
 func (st *Account) SetPendingNonce(nonce uint64) error {
+	if st.isMaxNonce() {
```
Comment (Liuhaai) [inline state/account.go:151]: the logic could be movd under switch

Comment (huangzhiran) [reply state/account.go:151]: if movd under switch. st.nonce will already ++ and maybe overflow already. pre-check maybe better

Comment (Liuhaai) [reply state/account.go:151]: Could you give an example?

**Review by Liuhaai — APPROVED** [review]

File: state/account.go (hunk 2 above)
Comment (huangzhiran) [reply state/account.go]: changed already

**Review by millken — APPROVED** [review]

**Pull request merged**